backstory/src/demo/java/com/google/sps/servlets/PerspectiveServlet.java

```diff
@@ -39,11 +45,94 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.setContentType("application/json;");
     Gson gson = new Gson();
 
-    String[] cannedOutput = new String[0];
+    // will hold text to analyze
+    String text;
 
-    // write canned output to servlet
-    String json = gson.toJson(cannedOutput);
+    String json = request.getReader().readLine();
 
-    response.getWriter().println(json);
+    // get the text from the JSON & handle error if it cannot be converted
+    try {
+      JSONObject jsonContainer = new JSONObject(json);
+      text = jsonContainer.getString("text");
+    } catch (JSONException exception) {
+      String errorMessage = "Could not convert text sent to server from JSON.";
+
+      handleError(response, HttpServletResponse.SC_BAD_REQUEST, errorMessage);
+      return;
+    }
+
+    // check that text is valid (not null or empty)
+    if (text == null || text.equals("")) {
+      handleError(response, HttpServletResponse.SC_BAD_REQUEST, "Text input was null or empty");
+      return;
+    }
+
+    // as we want to display the values as well as the decision in our demo
+    // we need to access internal code that the MVP would not
+    PerspectiveAPIFactory factory;
+    
+    try {
+      factory = new PerspectiveAPIFactoryImpl();
+    } catch (ClassNotFoundException | NoSuchMethodException | IllegalAccessException | InvocationTargetException exception) {
+      // if any errors were thrown for looking for api key, send that it was not retrieved back to page
+      handleError(response, HttpServletResponse.SC_INTERNAL_SERVER_ERROR, "Could not retrieve the Perspective API key.");
+      return;
+    }
+
+    PerspectiveAPI api = factory.newInstance();
+    PerspectiveAPIClient apiClient = new PerspectiveAPIClient(PerspectiveManager.getRequestedAttributes(), text);
```

Comment:
The above two lines look unneeded.

Reply:
They are needed to get values from PerspectiveAPI but somehow the relevant line about that was deleted. I will add that right back in, and group these together so that they make more sense.

Reply:
Done

Reply:
I see. This is fine, just as an FYI (no need to update at this time), an alternative would have been to add a perspectiveValues property to StoryDecision, so that you could use the manager directly.

Reply:
I see. I'll add that now.

Reply:
I did that, as mentioned below.

Reply:
Done